Report L2 status in get_info & oxend status

jagerman opened the pull request:
Currently it's not entirely obvious how to tell that your oxend is getting L2 updates properly; this commit aims to fix that.

This adds a `l2_tracker_height` field to get_info that reports the current L2 height via the node's l2 tracker, as an alternative to the existing `l2_tracker` field that reports the l2_height of the top block.

This new field is then used in the `oxend status` output to report the current L2 block heights of both the network and provider, such as:

    Height: 765478 (L2: 146073927/146074408) ON TESTNET, v11.2.0-bb1bf0fe5-dirty-dev(net v21), 8(out)+0(in) conns, up 18s
    SN: 5ec3d29f58335adae2b2989d635c3fd90e18b0ea17b441914d5c5da7441e0e62 not registered, proof: (never), last pings: NOT RECEIVED (storage), NOT RECEIVED (lokinet)
    BLS pk: 21fe4377778c0f14129cbdfd4c8f60987c8d871d34aad95f67727978af5da7c20e5a4413c885fbe604b58b4c3d3137674f4b2e0ff7dba4c6525984d88e54824b

I also shortened some of the strings on the status line to make it fit better in smaller terminals.

During testing of this, I encountered (and fixed) a somewhat obscure bug that causes interactive oxend commands to timeout if run on an uncontactable, deregistered, but not yet purged, service node, and so this updates the oxen-mq release with a bugfix version as well.

jagerman committed:
Report L2 status in get_info & oxend status

Currently it's not entirely obvious how to tell that your oxend is
getting L2 updates properly; this commit aims to fix that.

This adds a `l2_tracker_height` field to get_info that reports the
current L2 height via the node's l2 tracker, as an alternative to the
existing `l2_tracker` field that reports the l2_height of the top block.

This new field is then used in the `oxend status` output to report the
current L2 block heights of both the network and provider, such as:

    Height: 765478 (L2: 146073927/146074408) ON TESTNET, v11.2.0-bb1bf0fe5-dirty-dev(net v21), 8(out)+0(in) conns, up 18s
    SN: 5ec3d29f58335adae2b2989d635c3fd90e18b0ea17b441914d5c5da7441e0e62 not registered, proof: (never), last pings: NOT RECEIVED (storage), NOT RECEIVED (lokinet)
    BLS pk: 21fe4377778c0f14129cbdfd4c8f60987c8d871d34aad95f67727978af5da7c20e5a4413c885fbe604b58b4c3d3137674f4b2e0ff7dba4c6525984d88e54824b

I also shortened some of the strings on the status line to make it fit
better in smaller terminals.

jagerman committed:
Update oxen-mq

Update oxen-mq to fix a bug that could result in interactive mode
commands timing out on an unavailable, but unpurged, service node.

Doy-lee commented:
I find the notation `(L2: 146073927/146074408)` a little bit confusing because the LHS is always going to trail the RHS. LHS is the L2 height that's witnessed in to the work chain and the RHS is the L2 height, but it almost looks like a syncing indicator and that you'd want to see the numbers sync up.

Maybe `(L2 wit.: 146073927 top: 146074408)`? I know we're trying to keep the status line concise

jagerman commented:
> I find the notation `(L2: 146073927/146074408)` a little bit confusing because the LHS is always going to trail the RHS. LHS is the L2 height that's witnessed in to the work chain and the RHS is the L2 height, but it almost looks like a syncing indicator and that you'd want to see the numbers sync up.
> 
> Maybe `(L2 wit.: 146073927 top: 146074408)`? I know we're trying to keep the status line concise

How about:

    (L2 blk 146073927, cur +481)

because the precise values don't matter here so much as seeing a + to know you're ahead, and with a delta there it's much easier to tell that you are good or bad.  (I had to stare into the weeds for a couple seconds to figure out which of the 9-digit numbers was larger).

jagerman committed:
Tweak L2 height display

With an L2 tracker, this is now:

    (L2 blk: 146073927, cur: +481)

so that you can tell at a glance that your current L2 height is ahead of
the block (which is good).  + = good, - = bad.

Without an L2 tracker it's just:

    (L2: blk 146073927)

tewinget commented:
commit message typo (with -> without L2 tracker in second instance) in that tweak L2 height display commit, if you care.

jagerman committed:
Fix L2 display for pre-HF21 (e.g. current mainnet)

There is no block L2 height to show, and:

    (L2 blk: 0, cur: +331946115)

is not exactly useful.  This changes it to:

    (L2 cur: 331946115)

when there is no oxen-block L2 height.

Doy-lee commented:
Yeah that looks better, LGTM.
